compiler/config.py

```diff
@@ -144,13 +147,16 @@ def init_log_file():
 def read_vars_file(var_file_path):
     global config
 
+
     config['shell_variables'] = None
     config['shell_variables_file_path'] = var_file_path
     if(not var_file_path is None):
         vars_dict = {}
         with open(var_file_path) as f:
             lines = [line.rstrip() for line in f.readlines()]
 
+        # ??? MMG 2020-12-10 not sure how you're generating this file, but what about variables with newlines in them?
+        # MMG 2021-03-09 definitively breaking on IFS and function outputs (i.e., `declare -f`)
```

Comment:
Yes this is indeed very hacky, and we will probably need to fix this for smoosh tests to pass.